cli/module_generate.go

```diff
@@ -0,0 +1,726 @@
+	} else {
+		// If newModule.Namespace is a UUID/OrgID
+		org, err := resolveOrg(c, "", newModule.Namespace)
+		if err != nil {
+			return catchResolveOrgErr(cCtx, c, newModule, err)
+		}
+		newModule.OrgID = newModule.Namespace
+		newModule.Namespace = org.GetPublicNamespace()
+	}
+	return nil
+}
+
+func catchResolveOrgErr(cCtx *cli.Context, c *viamClient, newModule *moduleInputs, caughtErr error) error {
+	if strings.Contains(caughtErr.Error(), "not logged in") || strings.Contains(caughtErr.Error(), "error while refreshing token") {
+		originalWriter := cCtx.App.Writer
+		cCtx.App.Writer = io.Discard
+		err := c.loginAction(cCtx)
+		cCtx.App.Writer = originalWriter
+		if err != nil {
+			return err
+		}
+		return wrapResolveOrg(cCtx, c, newModule)
+	}
+	if strings.Contains(caughtErr.Error(), "none of your organizations have a public namespace") ||
+		strings.Contains(caughtErr.Error(), "no organization found for") {
+		return errors.Wrapf(caughtErr, "cannot create module for an organization of which you are not a member")
+	}
+	return caughtErr
+}
+
+// populateAdditionalInfo fills in additional info in newModule.
+func populateAdditionalInfo(newModule *moduleInputs) {
+	newModule.GeneratedOn = time.Now().UTC()
+	newModule.GeneratorVersion = version
+	newModule.ResourceSubtype = strings.Split(newModule.Resource, " ")[0]
+	newModule.ResourceType = strings.Split(newModule.Resource, " ")[1]
+
+	titleCaser := cases.Title(language.Und)
+	replacer := strings.NewReplacer("_", "", "-", "")
+	newModule.ModulePascal = replacer.Replace(titleCaser.String(newModule.ModuleName))
+	newModule.API = fmt.Sprintf("rdk:%s:%s", newModule.ResourceType, newModule.ResourceSubtype)
+	newModule.ResourceSubtypePascal = replacer.Replace(titleCaser.String(newModule.ResourceSubtype))
+	newModule.ModelPascal = replacer.Replace(titleCaser.String(newModule.ModelName))
+	newModule.ModelTriple = fmt.Sprintf("%s:%s:%s", newModule.Namespace, newModule.ModuleName, newModule.ModelName)
+}
+
+// Creates a new directory with moduleName.
+func setupDirectories(c *cli.Context, moduleName string) error {
+	debugf(c.App.Writer, c.Bool(debugFlag), "Setting up directories")
+	err := os.Mkdir(moduleName, 0o750)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func renderCommonFiles(c *cli.Context, module moduleInputs) error {
+	debugf(c.App.Writer, c.Bool(debugFlag), "Rendering common files")
+
+	// Render .viam-gen-info
+	infoBytes, err := json.MarshalIndent(module, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	infoFilePath := filepath.Join(module.ModuleName, ".viam-gen-info")
+	//nolint:gosec
+	infoFile, err := os.Create(infoFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s", infoFilePath)
+	}
+	defer utils.UncheckedErrorFunc(infoFile.Close)
+
+	if _, err := infoFile.Write(infoBytes); err != nil {
+		return errors.Wrapf(err, "failed to write generator info to %s", infoFilePath)
+	}
+
+	// Render workflows for cloud build
+	if module.EnableCloudBuild {
```

Comment:
(opt) this block of code gets pretty nested and that makes it pretty hard to follow. I think some refactoring could definitely be done to clean up the control flow here, but it might be a bit of a hassle to do. At a minimum though, I think we could move everything one level up by inverting this, i.e.,

```go
if !module.EnableCloudBuild {
    return
}
...
```

Reply:
I fell into a trap I hate. I purposefully didn't return early because this renders all common files, so I wanted to be able to leave space for additional common files that come after this. If we return early, that doesn't allow for adding new files afterward without changing the code again. 

But that's what you call "premature optimization"